### ISSUE TITLE
[V2] Schema types not available for completion and error highlighting

### ISSUE THREAD

**Issue by poqeteer** [jimkyndemeyer/js-graphql-intellij-plugin#242]
## Version and Environment Details
__Operation system__: macOs Mojave (10.14.4)

__IDE name and version__: WebStorm 2019.1 (March 21, 2019)

__Plugin version__: 2.0.0

## Expected Behaviour
Parser queries according to schema properly.

## Actual Behaviour
Displaying errors (on all fields) about unknown fields then suggesting an object of the exact same name.

## Steps to Reproduce / Link to Repo with Reproduction and Instructions
![image](https://user-images.githubusercontent.com/4682698/56007401-6c61be80-5c8d-11e9-8f76-3de16d1589e8.png)

Apologies for not being able to share schema or full query, this is a commercial product. The older version plugin didn't have this problem. I recently upgraded the version of the IDE and the plugin was not installed with the upgrade, so this is a fresh install of plugin.

**Comment by Karabur** [comment]
have same error. 
In my schema.json file it does not recognize any Type apart from Built-In primitives (String, Float etc.)

Any type which is defined in schema explicitly is marked as unknown, event Query, Mutation and Subscription. 

![image](https://user-images.githubusercontent.com/951973/56013172-228cce80-5cb5-11e9-9500-727486832ec5.png)

**Comment by jimkyndemeyer** [comment]
Hi.

Thanks for using the plugin.

I think this relates to using a JSON file as the schema path, but I haven't had time to verify this.

As a workaround you should be able to get this sorted by using GraphQL SDL to describe the schema, e.g. by letting the schema discovery process all GraphQL SDL in your project for a local schema, or by using an introspection to fetch your remote schema as described in https://github.com/jimkyndemeyer/graphql-config-examples/tree/master/remote-schema-introspection

I'll update once I know more about the underlying cause and a fix.

**Comment by jimkyndemeyer** [comment]
@poqeteer @Karabur So my theory that this was based on having a JSON schemaPath didn't hold up.

I realize that you might not be able to share your schemas due to commercial projects, but I'm unable to reproduce using the current information.

A minimal example to reproduce the issue might be required to progress, but before that please try the following:

- In the schemas panel, double click on the "Schema discovery summary" tree node and search for your root type, e.g. "Query", and select it. This should take you to where the schema discovery picked up the type. Is this part working as expected? Do you see the right number of types in the summary?
- Based on the GraphQL stack you're using, can you reproduce using their examples and link to it?

Thanks,
Jim.

**Comment by poqeteer** [comment]
@jimkyndemeyer I appreciate your dilemma... I'll see if I can get it to reproduce in a sample case... Unfortunately, in a bit of a crunch right now so will try to get back to this in a bit.

**Comment by sibelius** [comment]
I was having a problem like this, it was because my schemas was excluded, than I removed the exclusion and it worked well

**Comment by jimmyn** [comment]
After updateing to 2.1.0 I have all fields highlighted red and WebStrom offers me to each field to the field with the same name. Autocomplete works fine.
<img width="771" alt="Screen Shot 2019-04-22 at 14 28 23" src="https://user-images.githubusercontent.com/5391775/56498595-1271bc00-650b-11e9-835b-013bc74539a3.png">

**Comment by jimkyndemeyer** [comment]
@jimmyn If you clone https://github.com/jimkyndemeyer/graphql-config-examples and open it in webstorm, does it work? I expect this is related to project structure somehow, but I'll need a minimal example to reproduce before a fix can be found. Another thing to check is to use the "Invalidate caches and restart" option in the file menu. The index may have become invalid.

**Comment by jimmyn** [comment]
@jimkyndemeyer this is indeed related to project structure. I have a monorepo with the following structure

```
/frontend
/backend
schema.json
.graphqlconfig
```

.graphqlconfig
```
{
  "schemaPath": "./schema.json"
}
```

Everything works well in another project with a flat structure.

**Comment by sibelius** [comment]
@jimkyndemeyer can we make schema discover schemas that are excluded?

**Comment by jimkyndemeyer** [comment]
@jimmyn The issue might be that the content roots in the project don't include the schema, because the structure doesn't have to be flat. I'm off on vacation and won't be back for a week, so you may need to move/copy your schema and config file into the different directories to get this working until I have time to look at a fix.

@sibelius Not sure how to do that without potentially incurring a huge performance hit, which is why the scope is the "Project scope". You can include specific folders into the project scope, e.g. certain node packages, or you can configure additional "content roots" in the Idea/WebStorm project structure dialog.

**Comment by jtwozniak** [comment]
I encounter the same issue during removing apollo-codegen, after updating JSON schema generation as follows
 ```apollo-codegen introspect-schema``` to ```apollo schema:download```

The only difference in generated schemas was one prop that nests schema

This works:
```
{
  "data": {
    "__schema": { ...},
  }
}
```

and this doesn't
```
{
  "__schema": { ...},
  }
}
```

**Comment by jimkyndemeyer** [comment]
@jimmyn I've tried to setup up a structure that is similar to what you outlined, but I don't have any issues with schema discovery:

![image](https://user-images.githubusercontent.com/15837712/57127727-90e31f00-6d91-11e9-81ce-f1943568b68e.png)

Could you please attach a complete zipped project that demonstrates the issue.

**Comment by Karabur** [comment]
I can confirm - my schema generated with "_schema" as top level property and problem arises. 
```
{
  "__schema": {
    //schema here
  } 
}
```

but if I wrap it manually to "data" property - it starts to work fine with no errors, autocomplete works, etc. 
```
{
 "data":{
  "__schema": {
    //schema here
  } 
 }
}
```

**Comment by Karabur** [comment]
I've made minimal reproduce example.
[tmp-gql.zip](https://github.com/jimkyndemeyer/js-graphql-intellij-plugin/files/3143184/tmp-gql.zip)

if you goto ui/src/test.graphql file, it will show fields as errorneous:
![image](https://user-images.githubusercontent.com/951973/57158847-09ee7080-6daa-11e9-8ee6-766ceec5e2a7.png)


But if you go to shared/schema.json and wrap "__schema" to "data" field - it will start to work fine

![image](https://user-images.githubusercontent.com/951973/57158895-30aca700-6daa-11e9-9cb5-91422b4c5812.png)
![image](https://user-images.githubusercontent.com/951973/57158957-50dc6600-6daa-11e9-87d9-cbb182fe0b7a.png)

**Comment by jimkyndemeyer** [comment]
@Karabur Thanks for the reproduction. I've implemented a fix for the upcoming 2.1.1 release.

**Comment by jimkyndemeyer** [comment]
2.1.1 published, pending JetBrains approval.
